### PR TITLE
#2717 Move check of deprecated K8s versions to GitHub actions

### LINKED ISSUE #2717: Move check of deprecated K8s versions from TravisCI to GitHub Actions
Due to the current limitations of Travis-CI https://github.com/keptn/keptn/issues/2715, we are forced to switch some of our jobs from Travis-CI to GitHub Actions (which has no such limitations/restrictions right now).

As one of the first steps I would like to propose to switch the check for deprecated Kubernetes versions to GitHub Actions.

# Definition of Done

- [ ] Checks are removed from [travis-ci](https://github.com/keptn/keptn/blob/f89f31c5a5d95b80b4c933f9312b4598a1da92ee/.travis.yml#L284-L293)
- [ ] Verify if `travis-scripts/install_helm` and `travis-scripts/install_pluto` are still needed; remove them, if they are not needed anymore (i think `install_helm` is still needed).
- [ ] Kubernetes version check is implemented as a github action
- [ ] Branch protection rules in this repo are set to enforce this action

### PR DESCRIPTION
**Pull request by bacherfl**
Closes #2717

### CODE REVIEW

File: .github/workflows/deprecated-version-check.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: Check deprecated K8s versions
+on: [pull_request]
+jobs:
+  deprecated-version-check:
+    name: deprecated version check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code.
+        uses: actions/checkout@v1
```
Comment (christian-kreuzberger-dtx) [inline .github/workflows/deprecated-version-check.yml:9]: why do we not use checkout@v2?

File: test/assets/quality_gates_shipyard.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+apiVersion: "spec.keptn.sh/0.2.0"
```
Comment (christian-kreuzberger-dtx) [inline test/assets/quality_gates_shipyard.yaml:1]: this file is not needed for this PR, right?

File: gh-actions-scripts/install_helm.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
```
Comment (christian-kreuzberger-dtx) [inline gh-actions-scripts/install_helm.sh:2]: can we re-use the travis-scripts/install_helm.sh or is it different?

File: gh-actions-scripts/install_helm.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+# download
+curl https://get.helm.sh/helm-v3.2.4-linux-amd64.tar.gz > helm.tar.gz
+tar -zxvf helm.tar.gz
+if [ ! -d "$GITHUB_WORKSPACE/bin" ]; then
```
Comment (christian-kreuzberger-dtx) [inline gh-actions-scripts/install_helm.sh:6]: I guess I can answere it myself: the script is different, as we are using `$GITHUB_WORKSPACE` here.

File: test/assets/quality_gates_shipyard.yaml (hunk 2 above)
Comment (bacherfl) [reply test/assets/quality_gates_shipyard.yaml]: true, this is not needed for the PR. I have removed it

File: .github/workflows/deprecated-version-check.yml (hunk 1 above)
Comment (bacherfl) [reply .github/workflows/deprecated-version-check.yml]: i have updated it to v2

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM (please squash merge)

**Review by warber — APPROVED** [review]
lgtm

File: test/assets/quality_gates_shipyard.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (christian-kreuzberger-dtx) [reply test/assets/quality_gates_shipyard.yaml]: It's no longer in the PR (diff), it's just part of one of the commits, but we will squash-merge this PR

**Pull request merged**